### PR TITLE
Fix ldap authentication failed bug

### LINKED ISSUE #1325: Ldap Auth Failed when create user identified with authentication_ldap_simple
### Steps to reproduce the behavior (Required)
1. create user with authentication_ldap_simple
```sql
CREATE USER zhangsan IDENTIFIED WITH authentication_ldap_simple AS 'uid=zhansan,ou=company,dc=example,dc=com'
```

2. login 
```shell
mysql -uzhangsan -P9030 -h127.0.0.1 -p --default-auth mysql_clear_password --enable-cleartext-plugin
```

### Real behavior (Required)
Login failed

### StarRocks version (Required)
 - 1.19

### PR DESCRIPTION
**Pull request by gengjun-git**
Fix #1325 
Ldap authentication should use mysql_clear_password auth plugin. Add mysql_clear_password to supported plugins.

### CODE REVIEW

**Review by Astralidea — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/mysql/MysqlHandshakePacket.java [hunk 1]
```diff
@@ -34,7 +36,11 @@
     private static final MysqlCapability CAPABILITY = MysqlCapability.DEFAULT_CAPABILITY;
     // status flags not supported in StarRocks
     private static final int STATUS_FLAGS = 0;
-    private static final String AUTH_PLUGIN_NAME = "mysql_native_password";
+    private static final String DEFAULT_AUTH_PLUGIN_NAME = "mysql_native_password";
```
Comment (chaoyli) [inline fe/fe-core/src/main/java/com/starrocks/mysql/MysqlHandshakePacket.java:39]: Why to define two plugin name?

Comment (gengjun-git) [reply fe/fe-core/src/main/java/com/starrocks/mysql/MysqlHandshakePacket.java:39]: change

File: fe/fe-core/src/main/java/com/starrocks/mysql/MysqlHandshakePacket.java [hunk 2]
```diff
@@ -34,7 +36,11 @@
...
+    private static final ImmutableMap<String, Boolean> supportedPlugins = new ImmutableMap.Builder<String, Boolean>()
+            .put(DEFAULT_AUTH_PLUGIN_NAME, true)
+            .put("mysql_clear_password", true)
```
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/mysql/MysqlHandshakePacket.java:42]: why not define a static variable for this?

File: fe/fe-core/src/main/java/com/starrocks/mysql/MysqlHandshakePacket.java (hunk 1 above)
Comment (imay) [reply fe/fe-core/src/main/java/com/starrocks/mysql/MysqlHandshakePacket.java:39]: I think NATVIE_AUTH_PLUGIN_NAME is better

Comment (gengjun-git) [reply fe/fe-core/src/main/java/com/starrocks/mysql/MysqlHandshakePacket.java]: DONE

File: fe/fe-core/src/main/java/com/starrocks/mysql/MysqlHandshakePacket.java (hunk 2 above)
Comment (gengjun-git) [reply fe/fe-core/src/main/java/com/starrocks/mysql/MysqlHandshakePacket.java]: done

**Review by imay — APPROVED** [review]

**Review by Astralidea — APPROVED** [review]

**Pull request merged**